### ISSUE TITLE
@expo/vector-icons not showing after adding metro.config.js file in project root folder

### ISSUE THREAD

**Issue by francolivelli** [expo/expo#21568]
### Summary

Expo vector icons are not rendering after deployment.
As I'm using Expo Router, I'm using metro.
I installed `@expo/metro-config` and this is my metro.config.js file:

```
const { getDefaultConfig } = require("@expo/metro-config");

const defaultConfig = getDefaultConfig(__dirname);

module.exports = {
  resolver: {
    assetExts: [...defaultConfig.resolver.assetExts, "db"],
  },
};
```

Expected behaviour:
![image](https://user-images.githubusercontent.com/109685663/223214187-611456c2-5775-49e6-ab6f-7114b453c472.png)

What actually happens
![image](https://user-images.githubusercontent.com/109685663/223214030-fd88c8d2-1640-4878-8feb-dfbbc6415bc4.png)

### What platform(s) does this occur on?

Web

### SDK Version

48

### Environment

expo-env-info 1.0.5 environment info:
    System:
      OS: Linux 5.19 Ubuntu 22.04.1 LTS 22.04.1 LTS (Jammy Jellyfish)
      Shell: 5.1.16 - /bin/bash
    Binaries:
      Node: 16.17.0 - /usr/local/bin/node
      npm: 8.15.0 - /usr/local/bin/npm
    npmPackages:
      expo: ^48.0.4 => 48.0.5 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.71.3 => 0.71.3 
      react-native-web: ~0.18.7 => 0.18.12 
    npmGlobalPackages:
      expo-cli: 6.3.2
    Expo Workflow: managed

### Minimal reproducible example

https://github.com/francolivelli/expo-error

**Comment by Unknown-Coder769** [comment]
If you are experiencing issues with Expo vector icons not rendering after deployment, you may need to add some additional configuration to your project.

Firstly, make sure that you have installed react-native-vector-icons and expo-font packages. You can do this by running the following command in your project directory:
`npm install react-native-vector-icons expo-font`

ext, you will need to add the following code to your App.js file to load the icon fonts:
`import { Ionicons } from '@expo/vector-icons';
import { useFonts } from 'expo-font';

export default function App() {
  const [loaded] = useFonts({
    Ionicons: require('@expo/vector-icons/build/vendor/react-native-vector-icons/Fonts/Ionicons.ttf'),
  });

  if (!loaded) {
    return null;
  }

  // Your app code here
}`

Make sure to replace Ionicons with the name of the icon font that you want to load. You can find the names of the available icon fonts in the node_modules/@expo/vector-icons/build/vendor/react-native-vector-icons/Fonts directory.

Finally, you may need to add additional configuration to your metro.config.js file. Here's an example of how you can configure Metro to resolve icon fonts:
`const { getDefaultConfig } = require('@expo/metro-config');

const defaultConfig = getDefaultConfig(__dirname);

module.exports = {
  resolver: {
    assetExts: [...defaultConfig.resolver.assetExts, 'ttf', 'otf'],
  },
};`

This should resolve any issues you are experiencing with Expo vector icons not rendering after deployment.

**Comment by francolivelli** [comment]
Hi! I did everything you said but icons are still not rendering after deployment :(

**Comment by Unknown-Coder769** [comment]
Try clearing the cache of the application by typing ```expo r -c```. It can also be a possibility that your application is not bundled correctly. Also Check that the vector icons are included in the assets of your app.json file. Make sure that the assetBundlePatterns and assetPlugins are correctly set.

**Comment by francolivelli** [comment]
I really don't what's wrong. My app.json file is this:

`{
  "expo": {
    "scheme": "acme",
    "web": {
      "bundler": "metro"
    },
    "name": "cafe-tucan",
    "slug": "cafe-tucan",
    "assetBundlePatterns": ["assets/*"]
  }
}
`

Already cleared the cache but didn't work.

**Comment by Unknown-Coder769** [comment]
i'd suggest changing to the following code in the your app.json:

```"assetBundlePatterns": [ "**/*" ]"```

**Comment by Unknown-Coder769** [comment]
Then it is maybe a bundling problem. Try changing your bundler and if that doesn't work, then i dont know what will

**Comment by francolivelli** [comment]
It will be better to import little svg images as if they were icons lol

**Comment by JesusMaterano** [comment]
the problem is related to metro, im having the same issue, becuase im using expo-router which using metro to bundler

**Comment by WKampel** [comment]
Anyone figure this out?

**Comment by mackenziekira** [comment]
I am seeing this too after migrating to use metro to bundle web instead of webpack (and upgrading to Expo 48, plus switching to use EAS Updates).

In the browser console, I see the following warnings:
```
Failed to decode downloaded font: <URL>
OTS parsing error: invalid sfntVersion: 1008813135
```

**Comment by mackenziekira** [comment]
@WKampel for your issue, this thread might be helpful: https://github.com/expo/expo/issues/21802

**Comment by mackenziekira** [comment]
fwiw, the locally generated `/dist` folder contains the `Ionicons.ttf` file:

<img width="342" alt="Screen Shot 2023-04-17 at 11 58 36 AM" src="https://user-images.githubusercontent.com/7525998/232601851-4ecb99f3-0fdb-492e-a89f-a52c25d9a3df.png">

 So serving web locally works fine, e.g. via `npx serve dist`.

But when uploading `/dist` to vercel, somehow the entire `/node_modules` directory gets excluded:
<img width="267" alt="Screen Shot 2023-04-17 at 11 59 11 AM" src="https://user-images.githubusercontent.com/7525998/232601993-1eb4415e-cff5-464a-97e7-6ae6e89ba20a.png">

I think that's what's causing the browser console errors.

Our temporary workaround is to pull in the font and glyph files from `react-native-vector-icons` and manually generate the ionicons alongside some custom icons we added following this [blog](https://echobind.com/post/creating-custom-icon-fonts-in-react-native-and-expo) post.

**Comment by hyochan** [comment]
+1 for the issue. This only happens in the release web build.
I've tried to update all packages and remove `metro.config.js` but no luck 😥

**Comment by mogarick** [comment]
I'm having the same problem. I'm using Expo Router and when exporting for web the icons don't show up. I'm using expo version 48.0.16 and the error is still there. 

Here is an screen shot of my list folder
<img width="378" alt="image" src="https://user-images.githubusercontent.com/4414145/236986324-9db3834b-1440-4d10-bd18-df0efa8add83.png">

and here is how I see the page when running `npx serve --single` from de `dist` folder:

<img width="572" alt="image" src="https://user-images.githubusercontent.com/4414145/236986554-3b717664-7729-4b09-9f3f-0d41c49fda28.png">

@EvanBacon in which expo version is your fix #22094 gonna be applied? :)

**Comment by mogarick** [comment]
My current dirty workaround on `package.json` while the fix is released:
``` js
{
    "build": "npx expo export --platform web && pnpm fixmrepo",
    "fixmrepo": "cd dist && mv assets/__node_modules node_modules",
}
```

**Comment by jeltehomminga** [comment]
I'm having the exact same issue, locally when I build everything looks good and I can see the icons with npx expo start or npx serve dist. Only when I deploy to firebase all icons are gone. However if I look in my dist folder I do have a /node_modules folder, not __node_modules. So suggested workaround from @mogarick does not work for me

<img width="554" alt="CleanShot 2023-05-13 at 09 44 04@2x" src="https://github.com/expo/expo/assets/35220102/488fc097-c8c5-4621-a303-97f8c733f79c">


 any suggestions?

**Comment by mogarick** [comment]
> I'm having the exact same issue, locally when I build everything looks good and I can see the icons with npx expo start or npx serve dist. Only when I deploy to firebase all icons are gone. However if I look in my dist folder I do have a /node_modules folder, not __node_modules. So suggested workaround from @mogarick does not work for me
> 
> 
> <img width="554" alt="CleanShot 2023-05-13 at 09 44 04@2x" src="https://github.com/expo/expo/assets/35220102/488fc097-c8c5-4621-a303-97f8c733f79c">
> 
> 
> 
> 
> 
>  any suggestions?

@jeltehomminga you need to move node_modules as child of dist folder. In your image it is a child of assets. Try to move it up and check if that works.

**Comment by jeltehomminga** [comment]
> > I'm having the exact same issue, locally when I build everything looks good and I can see the icons with npx expo start or npx serve dist. Only when I deploy to firebase all icons are gone. However if I look in my dist folder I do have a /node_modules folder, not __node_modules. So suggested workaround from @mogarick does not work for me
> > <img alt="CleanShot 2023-05-13 at 09 44 04@2x" width="554" src="https://user-images.githubusercontent.com/35220102/238106068-488fc097-c8c5-4621-a303-97f8c733f79c.png">
> > any suggestions?
> 
> @jeltehomminga you need to move node_modules as child of dist folder. In your image it is a child of assets. Try to move it up and check if that works.

Thanks for the suggestion, have tried to do that, but unfortunately no luck  🤷🏼‍♂️

**Comment by hyochan** [comment]
Same here. Tried above https://github.com/expo/expo/issues/21568#issuecomment-1546702711 and no luck!

**Comment by hyochan** [comment]
> I'm having the exact same issue, locally when I build everything looks good and I can see the icons with npx expo start or npx serve dist. Only when I deploy to firebase all icons are gone. However if I look in my dist folder I do have a /node_modules folder, not __node_modules. So suggested workaround from @mogarick does not work for me
> 
> <img alt="CleanShot 2023-05-13 at 09 44 04@2x" width="554" src="https://user-images.githubusercontent.com/35220102/238106068-488fc097-c8c5-4621-a303-97f8c733f79c.png">
> any suggestions?

I am trying to investigate the problem because web is useless without icons..
I tested `dist` folder locally with `httpServer` and found that there is no problem with the path. When I normally build the `expo web` and run it, the icons are showing correctly.

The problem occurs when I deploy this. I see below error messages in console.
```
Failed to decode downloaded font: {font file path}/myfont.woff2

OTS parsing error: Failed to convert WOFF 2.0 font to SFNT

Failed to decode downloaded font: {font file path}/myfont.woff

OTS parsing error: incorrect file size in WOFF header

Failed to decode downloaded font: {font file path}/myfont.ttf

OTS parsing error: incorrect entrySelector for table directory
```
![Screenshot 2023-05-17 at 5 13 32 AM](https://github.com/expo/expo/assets/27461460/4e36f336-9507-43bd-91cf-66ff8fd7b4ca)

I just wanted to share quickly because I want others not to waste time with paths.

**Comment by hyochan** [comment]
I think this might help. https://www.youtube.com/watch?v=q2kEFf68BUo

I am still digging in for the solution.

**Comment by hyochan** [comment]
Finally solved it! I was deploying my webapp with firebase hosting and it ignores `node_modules` by default which we should contain.

I made a fix not to ignore `assets/node_modules` and everything is working again 🎉
https://dooboo.io

**Comment by hyochan** [comment]
Finally solved it! I was deploying my webapp with firebase hosting and it ignores `node_modules` by default which we should contain.

I made a fix not to ignore `assets/node_modules/**` and everything is working again 🎉

Take a look at working webapp if intersted.
https://app.dooboo.io

@jeltehomminga This info might help since you are also using firebase.

Also I think this issue can be closed now.

**Comment by ladyshaitan** [comment]
This is happening to me as well, only locally using metro for both android and ios.

**Comment by thisisashukla** [comment]
I am facing the same issue. When i am running my app on expo the icons are coming up fine but when i am building for production, all the vector icons are missing. Is there a solution for this?

**Comment by plebon** [comment]
Same issue for me. Using expo locally it works fine but when I build for Android the icons are not showing.

I tried expo versions 46.0.0, 47.0.0, 48.0.0 (with matching deps) and all had the same issue. It was definitely working previously with v46.0.0.

iOS builds and works as expected. It is only an issue on Android

**Comment by thisisashukla** [comment]
I have ditched MaterialCommunityIcons and now using Ionicons which comes bundled with expo. It is working for me now.

**Comment by rootedsoftware** [comment]
I fixed this issue on Android by importing the icon fonts I need in my App.js file.

```
const [fontsLoaded] = useFonts({
    ionicons: require('./assets/fonts/Ionicons.ttf'),
    'Material Design Icons': require('./assets/fonts/MaterialCommunityIcons.ttf'),
    'Font Awesome 5 Free Regular': require('./assets/fonts/FontAwesome5_Regular.ttf'),
    feather: require('./assets/fonts/Feather.ttf'),
    octicons: require('./assets/fonts/Octicons.ttf'),
  });
```

**Comment by joerush18** [comment]
> ### Summary
> Expo vector icons are not rendering after deployment. As I'm using Expo Router, I'm using metro. I installed `@expo/metro-config` and this is my metro.config.js file:
> 
> ```
> const { getDefaultConfig } = require("@expo/metro-config");
> 
> const defaultConfig = getDefaultConfig(__dirname);
> 
> module.exports = {
>   resolver: {
>     assetExts: [...defaultConfig.resolver.assetExts, "db"],
>   },
> };
> ```
> 
> Expected behaviour: ![image](https://user-images.githubusercontent.com/109685663/223214187-611456c2-5775-49e6-ab6f-7114b453c472.png)
> 
> What actually happens ![image](https://user-images.githubusercontent.com/109685663/223214030-fd88c8d2-1640-4878-8feb-dfbbc6415bc4.png)
> 
> ### What platform(s) does this occur on?
> Web
> 
> ### SDK Version
> 48
> 
> ### Environment
> expo-env-info 1.0.5 environment info: System: OS: Linux 5.19 Ubuntu 22.04.1 LTS 22.04.1 LTS (Jammy Jellyfish) Shell: 5.1.16 - /bin/bash Binaries: Node: 16.17.0 - /usr/local/bin/node npm: 8.15.0 - /usr/local/bin/npm npmPackages: expo: ^48.0.4 => 48.0.5 react: 18.2.0 => 18.2.0 react-dom: 18.2.0 => 18.2.0 react-native: 0.71.3 => 0.71.3 react-native-web: ~0.18.7 => 0.18.12 npmGlobalPackages: expo-cli: 6.3.2 Expo Workflow: managed
> 
> ### Minimal reproducible example
> https://github.com/francolivelli/expo-error

Simple Fix
Add @expo/vector-icons dependency on you package.json
npm install @expo/vector-icons

**Comment by Martijncvv** [comment]
> 

where and how should I implement this fix that Firebase doesn't ignore that specific path :P

**Comment by rrufus** [comment]
When closing and re-opening our app on iOS (not tested Android) we notice that occasionally the icons do not load (perhaps 10% of the time). This looks like a similar issue to this one so adding my comments here.

When this error occurs, our sentry.io captures an exception:

```
File 'file:///var/mobile/Containers/Data/Application/.../Library/Application%20Support/.expo-internal/assets/node_modules/@expo/vector-icons/build/vendor/react-native-vector-icons/Fonts/MaterialCommunityIcons.ttf' for font 'material-community' doesn't exist
```

I've seen the same error just for `MaterialIcons.ttf` too.

This is a managed expo project which uses eas build and has some custom native code:
expo | 49.0.6
@expo/vector-icons | 13.0.0
"expo-font": "~11.4.0"


We do not use a `useFonts` call in our app code, so I assume a `useFonts` call is done by the @expo/vector-icons library as otherwise I do not understand how the pasted error would include the "for font 'material-community'" line.

Does anyone have a solution or explanation?

**Comment by jj-apps** [comment]
> When closing and re-opening our app on iOS (not tested Android) we notice that **occasionally** the icons do not load (perhaps 10% of the time). This looks like a similar issue to this one so adding my comments here.
> 
> When this error occurs, our sentry.io captures an exception:
> 
> ```
> File 'file:///var/mobile/Containers/Data/Application/.../Library/Application%20Support/.expo-internal/assets/node_modules/@expo/vector-icons/build/vendor/react-native-vector-icons/Fonts/MaterialCommunityIcons.ttf' for font 'material-community' doesn't exist
> ```
> 
> I've seen the same error just for `MaterialIcons.ttf` too.
> 
> This is a managed expo project which uses eas build and has some custom native code: expo | 49.0.6 @expo/vector-icons | 13.0.0 "expo-font": "~11.4.0"
> 
> We do not use a `useFonts` call in our app code, so I assume a `useFonts` call is done by the @expo/vector-icons library as otherwise I do not understand how the pasted error would include the "for font 'material-community'" line.
> 
> Does anyone have a solution or explanation?
> 
> Ps, if useful, the app also fails to show any static images when this error described above occurs. Does this suggest that sometimes a file system has not completed being mounted / linked before the app reaches its next step of the boot process?

I have the same issue too. It only happens occasionally. I don't use useFonts, but use Font.loadAsync

Is there a solution to this problem yet?

**Comment by Scr3nt** [comment]
I fix the issue by removing expo insight you can see why here : https://github.com/expo/expo/issues/21885
@rrufus @jj-apps

**Comment by rrufus** [comment]
@Scr3nt this looks very promising, trying it now. THANK YOU!

**Comment by jj-apps** [comment]
> I fix the issue by removing expo insight you can see why here : #21885 @rrufus @jj-apps

I will try it 👍 

Thanks!

**Comment by AuroVarat** [comment]
<img width="311" alt="Screenshot 2023-11-13 at 12 05 16 AM" src="https://github.com/expo/expo/assets/6793316/1f855a96-3261-422f-954d-e6798a0b78dd">

@Martijncvv  You can remove the highlighted in the firebase.json line to let Firebase know not to ignore the node_modules in dist.

Removing this solves my issue and the icons appear in my Firebase deployed web app.

**Comment by tvillafane** [comment]
For some reason, I get a weird flicker when I use anything but Ionicons in my Expo web builds and it looks like my browser refetches a ton of assets (even my custom ttf fonts).  The comment by [thisisashukla](https://github.com/thisisashukla) seems to be a step in the right direction.

**Comment by robertherber** [comment]
Has anyone gotten around this in a smooth way on Cloudflare Pages? Haven't found a way to override the default behaviour of ignoring anything called ´node_modules`.

**Comment by leezaric** [comment]
> Finally solved it! I was deploying my webapp with firebase hosting and it ignores `node_modules` by default which we should contain.
> 
> I made a fix not to ignore `assets/node_modules/**` and everything is working again 🎉
> 
> Take a look at working webapp if interested. https://dooboo.io
> 
> @jeltehomminga This info might help since you are also using firebase.
> 
> Also I think this issue can be closed now.

It does work! I've struggled with this problem for days, finally, it's really to do with Firebase. Thanks.

**Comment by leezaric** [comment]
> <img alt="Screenshot 2023-11-13 at 12 05 16 AM" width="311" src="https://private-user-images.githubusercontent.com/6793316/282318698-1f855a96-3261-422f-954d-e6798a0b78dd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk2NDYwNTEsIm5iZiI6MTcwOTY0NTc1MSwicGF0aCI6Ii82NzkzMzE2LzI4MjMxODY5OC0xZjg1NWE5Ni0zMjYxLTQyMmYtOTU0ZC1lNjc5OGEwYjc4ZGQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDVUMTMzNTUxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTkwMzYwZDEzZGIyZGNlZjNlYzgxMmVmMjQwNjNhMWU4YzIxZWViZTZiZTk1YjdjZTQ5ODg5MmM3YjliN2RlMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.2gAjUzxcgGqyZn2ZlMc1GzvXWMUQHXfvvfD35UzkWno">
> @Martijncvv You can remove the highlighted in the firebase.json line to let Firebase know not to ignore the node_modules in dist.
> 
> Removing this solves my issue and the icons appear in my Firebase deployed web app.

It did work!

**Comment by robertherber** [comment]
> > Maybe there's a way to rewrite the path (i.e. rename it to something different than node_modules)?
> 
> @robertherber, I used the following script after `expo export -p web` to move assets from the node_modules directory and update URIs in the js bundle file. We have metro output defined as single, so this hack is fine. For other output targets, you need to modify the command to crawl each file.
> 
> ```
> #!/usr/bin/env bash
> mv dist/assets/node_modules/* dist/assets/
> find dist/_expo/static/js/web/ -type f -print0 | xargs -0 sed -i 's/assets\/node_modules/assets/g'
> ```

Thanks, it worked for me as well! 👍

**Comment by Donhv** [comment]
still have this issue 2024

**Comment by alexcittadini** [comment]
> > Maybe there's a way to rewrite the path (i.e. rename it to something different than node_modules)?
> 
> @robertherber, I used the following script after `expo export -p web` to move assets from the node_modules directory and update URIs in the js bundle file. We have metro output defined as single, so this hack is fine. For other output targets, you need to modify the command to crawl each file.
> 
> ```
> #!/usr/bin/env bash
> mv dist/assets/node_modules/* dist/assets/
> find dist/_expo/static/js/web/ -type f -print0 | xargs -0 sed -i 's/assets\/node_modules/assets/g'
> ```

I'm still having this issue.

I run npx expo export -p web then tried this code but get the following error on the third line (the find command). Any idea why or if I can manually replicate this?
sed: 1: "dist/_expo/static/js/we ...": extra characters at the end of d command

**Comment by zactwidale** [comment]
For those still having problems after the solution has been presented here, I have another puzzle piece that might help.

I tried the solution proposed by @hyochan and then clarifed more clearly by @AuroVarat.... It APPEARED not to work.  If you are still having problems after trying that solution then you might have the same issue I did.  

I was being served a stale broken version, not the fresh, fixed version. Ctl-Shift-R was the final puzzle piece that still took a while to find.

**Comment by niiashikwei** [comment]
FWIW:

I struggled with this but finally found a fix for my expo project that is deploying a web app using firebase. Here's how i resolved it:

1. I copied the "Ionicons.ttf" file into my "assets/fonts/" folder
2. Then preloaded the font in the main app file like so:

`    const [loaded] = useFonts({
        ionicons: require('../assets/fonts/Ionicons.ttf'), // Note that i used a lower case "i" here for "ionicons" - doesn't work when it's capitalized smh.
    });

    useEffect(() => {
        if (loaded) {
            SplashScreen.hideAsync();
        }
    }, [loaded]);

    if (!loaded) {
        return null;
    }`
    
    3. then i ran `npx expo export --platform web && firebase deploy`
    
    NB: I did not need to update my firebase.json to not ignore node-modules at all for this approach!

**Comment by Elue-dev** [comment]
> I fixed this issue on Android by importing the icon fonts I need in my App.js file.
> 
> ```
> const [fontsLoaded] = useFonts({
>     ionicons: require('./assets/fonts/Ionicons.ttf'),
>     'Material Design Icons': require('./assets/fonts/MaterialCommunityIcons.ttf'),
>     'Font Awesome 5 Free Regular': require('./assets/fonts/FontAwesome5_Regular.ttf'),
>     feather: require('./assets/fonts/Feather.ttf'),
>     octicons: require('./assets/fonts/Octicons.ttf'),
>   });
> ```

are the icons actually in your assets folder?

**Comment by Sourav12061999** [comment]
Could anybody give me a solution here? I am using s3 for deployment

**Comment by pachun** [comment]
@jypa 's script worked for me as well.

Just a quick reminder that ChatGPT is amazing; I'm also on a Mac and had the same problem as @alexcittadini. ChatGPT explained to me what the command was doing in plain English, why the command wasn't working correctly for me, how to make it work, and how to automate it on either my Mac or the Ubuntu machine that runs my GH action for deploying.

```
#!/bin/bash

# Check the operating system
if [[ "$OSTYPE" == "darwin"* ]]; then
  # MacOS: Use BSD sed
  find dist/_expo/static/js/web/ -type f -print0 | xargs -0 sed -i '' 's/assets\/node_modules/assets/g'
elif [[ "$OSTYPE" == "linux-gnu"* ]]; then
  # Linux: Use GNU sed
  find dist/_expo/static/js/web/ -type f -print0 | xargs -0 sed -i 's/assets\/node_modules/assets/g'
else
  echo "Unsupported OS: $OSTYPE"
  exit 1
fi
```

**Comment by micaelomota** [comment]
for anyone also deploying to firebase hosting, remember to put the headers specified here: https://docs.expo.dev/distribution/publishing-websites

```
 "headers": [
        {
          "source": "/**",
          "headers": [
            {
              "key": "Cache-Control",
              "value": "no-cache, no-store, must-revalidate"
            }
          ]
        },
        {
          "source": "**/*.@(jpg|jpeg|gif|png|svg|webp|js|css|eot|otf|ttf|ttc|woff|woff2|font.css)",
          "headers": [
            {
              "key": "Cache-Control",
              "value": "max-age=604800"
            }
          ]
        }
      ],
      ```
      
      you my also need to remove the node_modules from the ignore list in `firebase.json`

**Comment by ShadowMonarch01** [comment]
Here is the fix
`import { StatusBar } from 'expo-status-bar';
import RootStack from './src/screens/RootStack';
import { NavigationContainer } from '@react-navigation/native';
import { Ionicons, MaterialIcons, Entypo, AntDesign, Feather } from '@expo/vector-icons';
import { useFonts } from 'expo-font';

export default function App() {

  const [loaded] = useFonts({
    ...Ionicons.font,
    ...MaterialIcons.font,
    ...Entypo.font,
    ...AntDesign.font, 
    ...Feather.font
  });

  if (!loaded) {
    return null;
  }
  return (
    <NavigationContainer>
      <RootStack/>
      <StatusBar style="dark" />
    </NavigationContainer>
  );
}

`

**Comment by dbertolini** [comment]
If you are using Firebase for the hosting, a simple work around could be to remove the node_modules from the ignore. For example in your firebase.json file, in the hosting.ignore part, I removed node_modules.

"hosting": {
    "public": "public",
    "ignore": ["firebase.json", "**/.*"],        <----- here was the node_modules before
    "rewrites": [
      {
        "source": "**",
        "destination": "/index.html"
      }
    ]
  }, 
  
  Maybe this is not the right option, but at least it save me the day for a small proyect.

**Comment by scaryguy** [comment]
I spent 2-3 days because of this. God... Finally figure it out! 🎉 

The bottom line is that it was all because `node_modules` was in both `.gitignore` and `.dockerignore` files and it was causing font files (which live under node_modules folder within the public folder) to be NOT added to production build. **AND** I had to add an explicity copy to `Dockerfile`... 

In my case, I've been copy/pasting `dist` content to my Express app's `public/web` folder. And here are the things I had to do in order for ionicons and fontawesome to work properly.

`.gitignore` AND `.dockerignore`
```
# Of course you still want to keep this
node_modules

# But you MUST add an exclusion for the node_modules within your public
!apps/MyApp/public/web/assets/node_modules
```

`Dockerfile`
```
# Copy web assets first to ensure they're preserved
COPY public/web/assets/node_modules ./public/web/assets/node_modules/
```

I hope this saves some people from spending a few hours/days to debug...

**Comment by brentvatne** [comment]
closing this because the metro config posted in the original issue is incorrect:

```
const { getDefaultConfig } = require("@expo/metro-config");

const defaultConfig = getDefaultConfig(__dirname);

module.exports = {
  resolver: {
    assetExts: [...defaultConfig.resolver.assetExts, "db"],
  },
};
```

this doesn't actually ever export the default config

it should look like this:

```
const { getDefaultConfig } = require('expo/metro-config');

const config = getDefaultConfig(__dirname);

config.resolver.assetExts.push(
  // Adds support for `.db` files for SQLite databases
  'db'
);

module.exports = config;
```

see: https://docs.expo.dev/guides/customizing-metro/

we also warn in expo doctor when this happens!
